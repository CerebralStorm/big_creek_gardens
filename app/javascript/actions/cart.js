import Cookies from 'js-cookie'
import constants from '../constants'
ENV.Cookies = Cookies

let cartKey = function (currentUser = null) {
  if(!!currentUser) {
    return `${currentUser.id}_cart`;
  } else {
    return 'guest_cart';
  }
}

let verifyCart = function (currentUser = null, cart = null) {
  if(typeof(cart) !== 'object') {
    Cookies.set(cartKey(currentUser), {});
    cart = {}
  }
  return cart;
}

let loadCart = function (currentUser = null) {
  return function (dispatch) {
    let cart = Cookies.getJSON(cartKey(currentUser));
    cart = verifyCart(currentUser, cart)
    return dispatch({ type: constants.LOAD_CART, cart: cart })
  }
}

let updateCart = function (currentUser = null, cart = {}) {
  cart = verifyCart(currentUser, cart)
  Cookies.set(cartKey(currentUser), cart);
}

export { loadCart, updateCart };