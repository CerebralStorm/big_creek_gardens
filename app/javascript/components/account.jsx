import React from 'react'
import ReactDOM from 'react-dom'
import PropTypes from 'prop-types'

class Account extends React.Component {
  render() {
    return (
      <div className='container section'>
        <h1>My Account</h1>
      </div>
    );
  }
}

export default Account;