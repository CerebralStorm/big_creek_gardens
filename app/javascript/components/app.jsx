import React from 'react'
import ReactDOM from 'react-dom'
import PropTypes from 'prop-types'
import Header from './header'
import Footer from './footer'
import Index from './index'

import { library } from '@fortawesome/fontawesome-svg-core'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faShoppingCart } from '@fortawesome/free-solid-svg-icons'

library.add(faShoppingCart)

class App extends React.Component {
  render() {
    return (
      <div>
        <Header />
        {this.props.children}
        <Footer />
      </div>
    );
  }
}

export default App;