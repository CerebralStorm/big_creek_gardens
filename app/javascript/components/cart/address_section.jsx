import React from 'react';
import {CardElement} from 'react-stripe-elements';
import SelectUSState from 'react-select-us-states';

class AddressSection extends React.Component {
  render() {
    return (
      <div>
        <div className="form-group m-2">
          <label htmlFor="checkoutCity">Address</label>
          <input type="text" id='checkoutCity' className="form-control" />
        </div>
        <div className='row'>
          <div className="form-group m-2 col-sm-12 col-md-5">
            <label htmlFor="checkoutState">State</label>
            <SelectUSState id="checkoutState" className="form-control" />
          </div>
          <div className="form-group m-2 col-sm-12 col-md-5">
            <label htmlFor="checkoutZip">Zip</label>
            <input type="text" id='checkoutZip' className="form-control" />
          </div>
        </div>
      </div>
    );
  }
}

export default AddressSection;