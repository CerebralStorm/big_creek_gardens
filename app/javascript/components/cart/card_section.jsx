import React from 'react';
import {CardElement} from 'react-stripe-elements';

class CardSection extends React.Component {
  render() {
    return (
      <div className='col-12'>
        <div className="form-group m-2">
          <label htmlFor="checkoutCreditCard" className="col-sm-2 col-form-label">Credit Card</label>
          <CardElement id='checkoutCreditCard' className="form-control" />
        </div>
      </div>
    );
  }
}

export default CardSection;