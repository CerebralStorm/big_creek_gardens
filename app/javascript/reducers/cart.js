import constants from '../constants'

var cartReducer = function (state = {}, action) {
  switch (action.type) {
    case constants.LOAD_CART:
      console.log(state)
      console.log(action.cart)
      return action.cart;
    default:
      return state;
  }
}

export { cartReducer };