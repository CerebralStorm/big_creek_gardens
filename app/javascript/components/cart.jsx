import React from 'react'
import ReactDOM from 'react-dom'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { loadCart } from '../actions/cart'
import { Link } from 'react-router-dom'

import CartItem from './cart/cart_item'
import CheckoutForm from './cart/checkout_form'
import { Elements, StripeProvider } from 'react-stripe-elements';

class Cart extends React.Component {
  componentDidMount() {
    this.props.dispatch(loadCart())
  }

  render() {
    return (
      <div className="container cart-container">
        <div className="card shopping-cart">
          <div className="card-header bg-dark text-light">
            <i className="fa fa-shopping-cart" aria-hidden="true"></i>
            Shipping cart
            <Link to={'/shop'} className="btn btn-outline-success btn-sm pull-right">
              Continue Shopping
            </Link>
            <div className="clearfix"></div>
          </div>
          <div class="card-body">
            {Object.values(this.props.cart).map(cartItem => (
              <div>
                <CartItem key={cartItem.product.id} cartItem={cartItem} />
                <hr />
              </div>
            ))}
          </div>
        </div>
        <div class="card-footer">
          <div class="coupon col-md-5 col-sm-5 no-padding-left pull-left">
            <div class="row">
              <div class="col-6">
                <input type="text" class="form-control" placeholder="coupon code" />
              </div>
              <div class="col-6">
                <input type="submit" class="btn btn-default" value="Use coupon" />
              </div>
            </div>
          </div>
          <div class="pull-right" style={{margin: '10px'}}>
            <a href="" class="btn btn-success pull-right">Checkout</a>
            <div class="pull-right total-price" style={{margin: '5px'}}>
              <strong>Total price: <b>$50.00</b></strong>
            </div>
          </div>
          <div className='clearfix'></div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    currentUser: state.currentUser,
    cart: state.cart
  }
}

export default connect(mapStateToProps)(Cart)