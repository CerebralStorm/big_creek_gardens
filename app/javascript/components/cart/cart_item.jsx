import React from 'react'
import ReactDOM from 'react-dom'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { updateCart } from '../../actions/cart'

class CartItem extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="row">
        <div className="col-12 col-sm-12 col-md-2 text-center">
          <img className="img-responsive" src="http://placehold.it/120x80" alt="prewiew" width="120" height="80" />
        </div>
        <div className="col-12 text-sm-center col-sm-12 text-md-left col-md-6">
          <h4 className="product-name"><strong>{this.props.cartItem.product.name}</strong></h4>
          <h4><small>{this.props.cartItem.product.description}</small></h4>
        </div>
        <div className="col-12 col-sm-12 text-sm-center col-md-4 text-md-right row">
          <div className="col-3 col-sm-3 col-md-6 text-md-right line-item-price">
            <h6><strong>{this.props.cartItem.product.price} <span className="text-muted">x</span></strong></h6>
          </div>
          <div className="col-4 col-sm-4 col-md-4">
            <div className="quantity">
              <input type="button" value="+" className="plus" />
              <input type="number" step="1" max="99" min="1" value="1" title="Qty" className="qty" size="4" />
              <input type="button" value="-" className="minus" />
            </div>
          </div>
          <div className="col-2 col-sm-2 col-md-2 text-right">
            <button type="button" className="btn btn-outline-danger btn-xs">
              <i className="fa fa-trash" aria-hidden="true"></i>
            </button>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    currentUser: state.currentUser,
    cart: state.cart
  }
}

export default connect(mapStateToProps)(CartItem)

