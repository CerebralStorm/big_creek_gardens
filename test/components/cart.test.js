import React from 'react'
import { Cart } from '../../app/javascript/components/cart'

let cart = {};

let dispatch = jest.fn()

test('render a Cart', () => {
  const wrapper = shallow(<Cart cart={cart} dispatch={dispatch} />);
  expect(wrapper).toMatchSnapshot();
});