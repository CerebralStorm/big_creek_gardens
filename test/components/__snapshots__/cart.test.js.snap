// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a Cart 1`] = `
<div
  className="container cart-container"
>
  <div
    className="card shopping-cart"
  >
    <div
      className="card-header bg-dark text-light"
    >
      <i
        aria-hidden="true"
        className="fa fa-shopping-cart mr-md-3"
      />
      <strong>
        Shopping Cart
      </strong>
      <Link
        className="btn btn-outline-success btn-sm pull-right"
        replace={false}
        to="/shop"
      >
        Continue Shopping
      </Link>
      <div
        className="clearfix"
      />
    </div>
    <div
      className="card-body"
    />
  </div>
  <div
    className="card-footer"
  >
    <div
      className="coupon col-md-5 col-sm-5 no-padding-left pull-left"
    >
      <div
        className="row"
      >
        <div
          className="col-6"
        >
          <input
            className="form-control"
            placeholder="coupon code"
            type="text"
          />
        </div>
        <div
          className="col-6"
        >
          <input
            className="btn btn-default"
            type="submit"
            value="Use coupon"
          />
        </div>
      </div>
    </div>
    <div
      className="pull-right"
      style={
        Object {
          "margin": "10px",
        }
      }
    >
      <Link
        className="btn btn-success pull-right"
        replace={false}
        to="/checkout"
      >
        Checkout
      </Link>
      <div
        className="pull-right total-price"
        style={
          Object {
            "margin": "5px",
          }
        }
      >
        <strong>
          Total price: 
          <b>
            $
            0.00
          </b>
        </strong>
      </div>
    </div>
    <div
      className="clearfix"
    />
  </div>
</div>
`;
